gcc/cp/contracts.cc

```diff
@@ -1439,9 +1440,14 @@ build_contract_condition_function (tree fndecl, bool pre)
       DECL_WEAK (fn) = false;
       DECL_COMDAT (fn) = false;
 
-      /* We haven't set the comdat group on the guarded function yet, we'll add
-	 this to the same group in comdat_linkage later.  */
-      //gcc_assert (!DECL_ONE_ONLY (fndecl));
+      /* We may not have set the comdat group on the guarded function yet.
```

Comment:
this one should already be present - in the merged upstream from earlier today - since you applied it upstream, right?

Reply:
it's not present in this branch yet, but I presume it will be at some point. Removed the patch from this commit as it's not needed and will be picked up from trunk eventually.